tools/projmgr/docs/Manual/Overview.md

```diff
@@ -0,0 +1,1317 @@
+`target-types:`  | `*.csolution.yml`                              | Start of the [Target type declaration list](#target-and-build-types) that allow to switch between [different targets](#project-setup-for-multiple-targets-and-test-builds).
+`build-types:`   | `[defaults].csettings.yml`, `*.csolution.yml`  | Start of the [Build type declaration list](#target-and-build-types) that allow to switch between different build settings such as: Release, Debug, Test.
+`groups:`        | `*.cproject.yml`, `*.clayer.yml`               | Start of a list that adds [source groups and files](#groups-and-files) to a project or layer.
+`layers:`        | `*.cproject.yml`                               | Start of a list that adds software layers to a project.
+`components:`    | `*.cproject.yml`, `*.clayer.yml`               | Start of a list that adds software components to a project or layer.
+
+
+## General Properties
+
+The keywords below can be used at various levels in this file types: `[defaults].csettings.yml`, `*.csolution.yml`, and `*.cproject.yml`. 
+
+Keyword         | Description
+:---------------|:------------------------------------
+`compiler:`     | Selection of the toolchain used for the project, i.e. `GCC`, `AC6`, `IAR`, optional with version, i.e AC6@6.16-LTS
+`device:`       | [Unique device name](#device-name-conventions), optionally with vendor and core.  A device is derived from the `board:` setting, but `device:` overrules this setting.
+`board:`        | [Unique board name](#board-name-conventions), optionally with vendor.
+`optimize:`     | Generic optimize levels (max, size, speed, debug), mapped to the toolchain by CMSIS-Build.
+`debug:`        | Generic control for the generation of debug information (on, off), mapped to the toolchain by CMSIS-Build.
+`warnings:`     | Control warnings (could be: no, all, Misra, AC5-like), mapped to the toolchain by CMSIS-Build.
+`defines:`      | [#define symbol settings](#defines) for the code generation tools.
+`undefines:`    | [Remove #define symbol settings](#undefines) for the code generation tools.
+`add-paths:`    | [Add include path settings](#add-paths) for the code generation tools.
+`del-paths:`    | [Remove include path settings](#del-paths) for the code generation tools.
+`misc:`         | [Miscellaneous literal tool-specific controls](#misc) that are passed directly to the tools depending on the file type.
+
+**Notes:**
+
+- `defines:`, `add-paths:`  and `misc:` are additive. All other keywords overwrite previous settings.
+
+
+
+## Target and Build Types
+
+The section [Project setup for multiple targets and test builds](#project-setup-for-multiple-targets-and-test-builds) describes the concept of  `target-types` and `build-types`.  These *types* can be defined in the following files in the following order:
+
+1. `default.csettings.yml`  where it defines global *types*, such as *Debug* and *Release* build.
```

Comment:
I still think it could be good to have a user level file, shared across solutions.